scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -221,17 +222,35 @@ class FossId(
                         .checkResponse("create project")
                 }
 
-                log.info { "Creating scan for URL $url and revision $revision ..." }
+                val scans = service.listScansForProject(user, apiKey, projectCode)
+                    .checkResponse("list scans for project").data
+                requireNotNull(scans)
 
-                val scanCode = createScan(projectCode, url, revision)
-
-                log.info { "Initiating download from Git repository ..." }
+                val existingScan = scans.sortedByDescending { scan -> scan.id }.find { scan ->
```

Comment:
Nit: For one-liners / when there is no variable shadowing, we usually stick to `it`, i.e. change `scan -> scan.id` to just `it.id`.